bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -371,6 +371,11 @@ func (r *KubeadmConfigReconciler) refreshBootstrapTokenIfNeeded(ctx context.Cont
 
 	secret, err := getToken(ctx, remoteClient, token)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Error(err, "token secret is gone, triggering creation of new token")
```

Comment:
```suggestion
			log.Error(err, "Bootstrap token secret is gone, triggering creation of new token")
```

Also, big nit, but should this really be an error log? Even if this not part of the normal flow, if we have a way to recover from it, should we surface it as an error? My concern is if this would be confusing from someone looking over the logs.

Reply:
Agree, and done. I had initially made it an error since it's rare, but that's admittedly not a good reason.